daceml/onnx/op_implementations/pure_implementations.py

```diff
@@ -499,23 +499,58 @@ def prog(X, Y):
         return program_for_node(prog, sdfg, state, node)
 
 
+
 @op_implementation(op="Reshape", name="pure")
 class PureReshape(ONNXForward):
+    '''
+        Reshape expansion: this relies on views
+    '''
     @staticmethod
     def forward(node: onnx_op.ONNXOp, state: SDFGState,
                 sdfg: SDFG) -> typing.Union[Node, SDFG]:
-        new_shape = out_desc_with_name(node, state, sdfg, "reshaped").shape
-        node.remove_in_connector("shape")
+        node.validate(sdfg, state)
```

Comment:
nit: this is already called before the expansion happens, so it's also not necessary